scripts/make_summary.py

```diff
@@ -540,77 +540,21 @@ def calculate_prices(n, label, prices):
 
 
 def calculate_weighted_prices(n, label, weighted_prices):
-    # Warning: doesn't include storage units as loads
-
-    weighted_prices = weighted_prices.reindex(
-        pd.Index(
-            [
-                "electricity",
-                "heat",
-                "space heat",
-                "urban heat",
-                "space urban heat",
-                "gas",
-                "H2",
-            ]
-        )
-    )
-
-    link_loads = {
-        "electricity": [
-            "heat pump",
-            "resistive heater",
-            "battery charger",
-            "H2 Electrolysis",
-        ],
-        "heat": ["water tanks charger"],
-        "urban heat": ["water tanks charger"],
-        "space heat": [],
-        "space urban heat": [],
-        "gas": ["OCGT", "gas boiler", "CHP electric", "CHP heat"],
-        "H2": ["Sabatier", "H2 Fuel Cell"],
-    }
-
-    for carrier, value in link_loads.items():
-        if carrier == "electricity":
-            suffix = ""
-        elif carrier[:5] == "space":
-            suffix = carrier[5:]
-        else:
-            suffix = " " + carrier
-
-        buses = n.buses.index[n.buses.index.str[2:] == suffix]
 
-        if buses.empty:
-            continue
-
-        if carrier in ["H2", "gas"]:
-            load = pd.DataFrame(index=n.snapshots, columns=buses, data=0.0)
-        else:
-            load = n.loads_t.p_set[buses.intersection(n.loads.index)]
-
-        for tech in value:
-            names = n.links.index[n.links.index.to_series().str[-len(tech) :] == tech]
+    carriers = n.buses.carrier.unique()
 
-            if not names.empty:
-                load += (
-                    n.links_t.p0[names].T.groupby(n.links.loc[names, "bus0"]).sum().T
-                )
-
-        # Add H2 Store when charging
-        # if carrier == "H2":
-        #    stores = n.stores_t.p[buses+ " Store"].groupby(n.stores.loc[buses+ " Store", "bus"],axis=1).sum(axis=1)
-        #    stores[stores > 0.] = 0.
-        #    load += -stores
+    for carrier in carriers:
+        grouper = n.statistics.groupers.get_bus_and_carrier
+        balance = n.statistics.energy_balance(
```

Comment:
you can directely use withdrawal, then you do not need to wilter for values <0

Reply:
done!